wgpu-core/src/device/mod.rs

```diff
@@ -1521,30 +1521,37 @@ impl<A: HalApi> Device<A> {
                 }
                 Br::Sampler(id) => {
                     match decl.ty {
-                        wgt::BindingType::Sampler {
-                            filtering,
-                            comparison,
-                        } => {
+                        wgt::BindingType::Sampler(ty) => {
                             let sampler = used
                                 .samplers
                                 .use_extend(&*sampler_guard, id, (), ())
                                 .map_err(|_| Error::InvalidSampler(id))?;
 
-                            // Check the actual sampler to also (not) be a comparison sampler
-                            if sampler.comparison != comparison {
-                                return Err(Error::WrongSamplerComparison {
-                                    binding,
-                                    layout_cmp: comparison,
-                                    sampler_cmp: sampler.comparison,
-                                });
+                            // Allowed sampler values for filtering and comparison
+                            let (allowed_filtering, allowed_comparison) = match ty {
```

Comment:
since `allowed_comparison` is always `Some`, why does it need to be an `Option`?

Reply:
Good point, pulled out the `Option`.